### ISSUE TITLE
Systemd service and tempfiles config are not installed.

### ISSUE THREAD

**Issue by loqs** [wesnoth/wesnoth#5692]
### Game and System Information
 - What version of the game are you running?
 1.15.11+dev
 - Where did you download it from? (Steam, Mac App Store, our website, built from source, etc.)
 git://github.com/wesnoth/wesnoth.git
 - If you're using a custom build, what compiler did you use, and what commit did you build at?
  * gcc 10.2.0
  * cmake 3.19.6
  * make 4.3
  * systemd 247
  * a2c795bfd7a22c6196f2b5bd1386edaf1523e541
 - What OS are you running?
Arch linux

### Describe the bug
Systemd unit file and tmpfiles config are not installed when systemd support is enabled with scons.

### To Reproduce
```
scons prefix=/usr docdir=/usr/share/doc/wesnoth/html/ fifodir=/run/wesnothd systemd=yes
scons install prefix=/usr docdir=/usr/share/doc/wesnoth/html/ fifodir=/run/wesnothd systemd=yes
```
### Expected behavior
The systemd unit file should be installed to /usr/lib/systemd/system/wesnothd.service.
The tempfiles config should be installed to /usr/lib/tmpfiles.d/wesnothd.conf.

### Screenshots
NA

### Savefiles
NA

### Additional context
Systemd is not detected as the systemd binary is installed to /usr/lib/systemd/systemd but passing the option overrides the detection failure.
Specifying the files as targets causes them to be generated and then they are installed as expected.
```
scons prefix=/usr docdir=/usr/share/doc/wesnoth/html/ fifodir=/run/wesnothd systemd=yes packaging/systemd/wesnothd.service packaging/systemd/wesnothd.conf wesnoth wesnothd
scons install prefix=/usr docdir=/usr/share/doc/wesnoth/html/ fifodir=/run/wesnothd systemd=yes
```
The failure appears to be because in the top level SConstruct the files are installed by:
```
if env["systemd"]:
    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.service", "lib/systemd/system")
    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.conf", "lib/tmpfiles.d")
```
However the env.InstallData calls do not install the files because wesnothd.service and wesnothd.conf have not been generated from their .in files.

I tested if this was the issue by adding an additional optional argument to InstallData to allow files to be absent when they are to be generated and that resolved the issue but may not be the right approach.
```diff
diff --git a/SConstruct b/SConstruct
index d969e9010e6..b3f1f72be30 100755
--- a/SConstruct
+++ b/SConstruct
@@ -747,8 +747,8 @@ if not access(fifodir, F_OK):
     AlwaysBuild(fifodir)
     env.Alias("install-wesnothd", fifodir)
 if env["systemd"]:
-    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.service", "lib/systemd/system")
-    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.conf", "lib/tmpfiles.d")
+    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.service", "lib/systemd/system", generated=True)
+    env.InstallData("prefix", "wesnothd", "#packaging/systemd/wesnothd.conf", "lib/tmpfiles.d", generated=True)
 
 # Wesnoth campaign server
 env.InstallBinary(campaignd)
diff --git a/scons/install.py b/scons/install.py
index 28b2c4dd8e6..69641688ffe 100644
--- a/scons/install.py
+++ b/scons/install.py
@@ -65,7 +65,7 @@ def InstallBinary(env, source):
         env.InstallAs(os.path.join(installdir, binary + env["program_suffix"]), source)
     )
 
-def InstallData(env, datadir, component, source, subdir = ""):
+def InstallData(env, datadir, component, source, subdir = "", generated=False):
     installdir = Dir(env.subst(os.path.join(env["destdir"], env[datadir].lstrip("/"), subdir)))
     sources = map(Entry, Flatten([source]))
     dirs = []
@@ -73,7 +73,7 @@ def InstallData(env, datadir, component, source, subdir = ""):
         if isinstance(source, SCons.Node.FS.Dir) or source.isdir():
             dirs.append(source)
         else:
-            if source.exists():
+            if source.exists() or generated==True:
                 env.Alias("install-" + component, env.Install(installdir, source))
     if dirs:
         if len(dirs) == 1:
```
Please also consider using the systemctl binary instead of the systemd binary to detect systemd's presence as the former should be on standard search paths or add /usr/lib/systemd to the paths to search for the systemd binary.

**Comment by loonycyborg** [comment]
I think entire `if source.exists()`  should be dropped with body executed unconditionally. Not sure that the `InstallData` silently ignoring non-existent sources is a useful behavior.

**Comment by soliton-** [comment]
Is this resolved then?

**Comment by loqs** [comment]
Fixed by 546355f32e907a35d6d18b7784f2dface962c180 and 5d8c47639611185ea19fda1d8a879d0a388cbe1c
